internal/oci/oci_linux.go

```diff
@@ -85,8 +85,12 @@ func (r *runtimeOCI) containerStats(ctr *Container, cgroup string) (*ContainerSt
 	stats.PIDs = cgroupStats.Pids.Current
 	stats.BlockInput, stats.BlockOutput = calculateBlockIO(cgroupStats)
 
-	if ctr.state != nil {
-		netNsPath := fmt.Sprintf("/proc/%d/ns/net", ctr.state.Pid)
+	// Try our best to get the net namespace path.
+	// If pid() errors, the container has stopped, and the /proc entry
+	// won't exist anyway.
+	pid, _ := ctr.pid() // nolint:errcheck
+	if pid > 0 {
+		netNsPath := fmt.Sprintf("/proc/%d/ns/net", pid)
```

Comment:
Why don't we return the pinned path here? We can get rid of the pid check then.

Reply:
a couple of reasons:

primarily: we don't have direct access to the Sandbox object, which owns the namespaces. We'd have to ask the server what the sandbox is, and then ask  the sandbox what the namespace is.
secondarily, is it possible to have a private net namespace? if so, the sandbox level namespace won't be the correct one

Reply:
Atleast for the pinned ns case.

Reply:
There is no private net namespaces in a pod. All the containers always share the pod network namespace.

Reply:
ah, well we still have the access problem. I would replumb this as a follow up, if we'd like